### PR TITLE
Deprecate Osmoutils, Osmomath

### PR DESCRIPTION
**Pull request by mattverse**
Closes: #XXX

## What is the purpose of the change

> Add a description of the overall background and high level changes that this PR introduces

*(E.g.: This pull request improves documentation of area A by adding ....*

## Testing and Verifying

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added unit test that validates ...*
  - *Added integration tests for end-to-end deployment with ...*
  - *Extended integration test for ...*
  - *Manually verified the change by ...*

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: osmomath/decimal.go [hunk 1]
```diff
@@ -1025,15 +1025,15 @@ func MaxBigDec(d1, d2 BigDec) BigDec {
 // DecEq returns true if two given decimals are equal.
 // Intended to be used with require/assert:  require.True(t, DecEq(...))
 //
-//nolint:thelper
+//nolint:thelper, revive
```
Comment (nicolaslara) [inline osmomath/decimal.go:1028]: what does this do?

File: osmoutils/accum/errors.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ import (
 )
 
 var (
-	ZeroSharesError = errors.New("shares must be non-zero")
+	ErrZeroShares = errors.New("shares must be non-zero")
```
Comment (nicolaslara) [inline osmoutils/accum/errors.go:11]: This seems unrelated. It's not a problem, just don't see why it needed to be changed in this pr

File: x/ibc-hooks/go.mod [hunk 3]
```diff
@@ -3,16 +3,15 @@ module github.com/osmosis-labs/osmosis/x/ibc-hooks
 go 1.21
 
 require (
-	cosmossdk.io/errors v1.0.0
-	github.com/CosmWasm/wasmd v0.40.1
+	cosmossdk.io/errors v1.0.1
+	github.com/CosmWasm/wasmd v0.45.1-0.20231128163306-4b9b61faeaa3
```
Comment (nicolaslara) [inline x/ibc-hooks/go.mod:7]: why do we need to specify a commit here?

File: osmoutils/sumtree/tree.pb.go [hunk 4]
```diff
@@ -167,26 +167,26 @@ func init() {
 func init() { proto.RegisterFile("osmosis/store/v1beta1/tree.proto", fileDescriptor_72b0b7af579d13be) }
 
 var fileDescriptor_72b0b7af579d13be = []byte{
-	// 297 bytes of a gzipped FileDescriptorProto
```
Comment (nicolaslara) [inline osmoutils/sumtree/tree.pb.go:170]: why are these changing? (it's happening in a few protos)

File: osmoutils/accum/errors.go (hunk 2 above)
Comment (mattverse) [reply osmoutils/accum/errors.go:11]: Lint problem. Basically by bringing osmoutils / osmomath into our repo, it started to get linted following our existing linting rules. Got bunch of linting problems that I needed to fix

File: osmomath/decimal.go (hunk 1 above)
Comment (mattverse) [reply osmomath/decimal.go:1028]: allows 5+ return arguments. (by bringing osmoutils / osmomath into our repo, it started to get linted following our existing linting rules. Got bunch of linting problems that I needed to fix)

File: osmoutils/sumtree/tree.pb.go (hunk 4 above)
Comment (mattverse) [reply osmoutils/sumtree/tree.pb.go:170]: probably because our proto gen docker file has been changed since this proto file was last generated . ( I needed to regenerate proto files due to our import paths for osmomath, osmoutils changing)

File: x/ibc-hooks/go.mod (hunk 3 above)
Comment (mattverse) [reply x/ibc-hooks/go.mod:7]: Automatic `go mod tidy` replaces that happens, note that these are not the actual dep updates happening with us replacing the dep version / commits in the `replace` section below

**Comment by mattverse** [comment]
Closing this PR per disuccsion with @nicolaslara towards a better design
